core/src/main/scala/org/http4s/blaze/channel/nio2/ByteBufferHead.scala

```diff
@@ -106,23 +106,25 @@ private[nio2] final class ByteBufferHead(channel: AsynchronousSocketChannel, buf
     case e: Throwable => super.checkError(e)
   }
 
-  override protected def closeWithError(t: Throwable): Unit = {
+  override protected def doClosePipeline(cause: Option[Throwable]): Unit = {
     val needsClose = synchronized {
       val reason = closeReason
-      if (reason == null || reason == EOF) {
-        closeReason = t
+      reason match {
+        case None | Some(EOF) =>
+          closeReason = Some(cause.getOrElse(EOF))
+        case _ => // nop
       }
-      reason == null
+      reason.isEmpty
     }
 
-    t match {
-      case EOF => logger.debug(s"closeWithError(EOF)")
-      case t => logger.error(t)("NIO2 channel closed with an unexpected error")
+    cause match {
+      case Some(t) => logger.error(t)("NIO2 channel closed with an unexpected error")
+      case None => logger.debug(s"closeWithError(EOF)")
```

Comment:
This scaladoc seems iffy now: it's not the name of the method nor the argument anymore.